wordpress-mobile/WordPress-Android
# Replaces bundle usage in the editor with a local database

Partially #9685 and #19157

## Description
This PR tries to mitigate the `TransactionTooLargeException` error in the editor by replacing `Bundle` for `Parcelable` objects with a custom database implementation.

-----

## To Test:
* Make a sanity check in the editor
* Verify that the information is stored and retrieved correctly from the database on configuration change
  1. Open the editor
  2. Make add some changes
  3. Put the app in the background and reopen
  4. Change the device orientation
  5. **Verify** that the app behaves as expected
* Verify that the revision is retrieved correctly from history
  1. Open the editor
  2. Make add some changes and save
  3. Make some more changes and save
  4. Open the option menu and select _History_
  5. Select the desired revision
  6. Tap _Load_ on the code editor
  7. **Verify** that the correct revision is loaded

-----

## Regression Notes

1. Potential unintended areas of impact

    Editor

8. What I did to test those areas of impact (or what existing automated tests I relied on)

    Manual testing

9. What automated tests I added (or what prevented me from doing so)

    Added tests for the added local db

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
